packages/taco/src/conditions/context/context.ts

```diff
@@ -178,6 +179,26 @@ export class ConditionContext {
         }
       }
     }
+    // If it's a JSON API condition, check url and query
+    if (condition.conditionType === JsonApiConditionType) {
+      const urlComponents = condition.endpoint.replace("https://", "").split("/");
+      for (const param of urlComponents ?? []) {
+        if (this.isContextParameter(param)) {
+          requestedParameters.add(param);
+        }
+      }
+      if (condition.query) {
+        const queryParams = condition.query.match(":[a-zA-Z_]+");
```

Comment:
There is an existing regex for context variables - `CONTEXT_PARAM_REGEXP` in `const.ts` that you can use. You should use that one instead. Context parameters allow numbers to be in the string (just not the first character).

Reply:
@theref discovered some nuance in the existing regex (`CONTEXT_PARAM_REGEXP`) that does not fully cover the Json API condition use case.  In particular the `CONTEXT_PARAM_REGEXP` enforces that the pattern must match at the start of a string (`^`), however, some situations a context parameter may appear as a substring in the middle of an endpoint (for example `https://example.com/:userAddress`).  We think this is specific to the case of JSON API conditions, but let us know if you think otherwise.

Reply:
I'd also like to point out that there is a small bug in `CONTEXT_PARAM_REGEXP `: `^:[a-zA-Z_][a-zA-Z0-9_]*$`.  The usage of the `*` character here permits matching zero tokens, which means that a simple `:` will match positively.

Reply:
There are two context regexes in `const.ts`.
- CONTEXT_PARAM_FULL_MATCH_REGEXP - matches a regex without expecting it to be at the beginning of the string i.e. no caret.
- CONTEXT_PARAM_FULL_MATCH_REGEXP - expects the entire string to be a context variable

I believe (?) the first one should be sufficient here.

Reply:
> It's subtle, but the [a-zA-Z_] part (first block) ensures that the context variable starts with a letter or underscore, the rest of regex allows the remaining characters to be a letter, number, or underscore.

Ah yes, you are correct!

Reply:
> There are two context regexes in `const.ts`.
> 
> * CONTEXT_PARAM_REGEXP - matches a regex without expecting it to be at the beginning of the string i.e. no caret.
> * CONTEXT_PARAM_FULL_MATCH_REGEXP - expects the entire string to be a context variable
> 
> I believe (?) the first one should be sufficient here.

Hmm, perhaps it's in another branch?  I don't see two, only `CONTEXT_PARAM_REGEXP` in `const.ts`

Reply:
https://github.com/nucypher/taco-web/blob/epic-v0.6.x/packages/taco/src/conditions/const.ts#L10-L15

Reply:
Yep - that looks suitable.